extensions/grpc/deployment/src/main/java/io/quarkus/grpc/deployment/GrpcClientBuildItem.java

```diff
@@ -0,0 +1,81 @@
+package io.quarkus.grpc.deployment;
+
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+import org.jboss.jandex.DotName;
+
+import io.quarkus.builder.item.MultiBuildItem;
+
+public final class GrpcClientBuildItem extends MultiBuildItem {
+
+    private final String name;
+    private final Set<StubInfo> stubs;
+
+    public GrpcClientBuildItem(String name) {
+        this.name = name;
+        this.stubs = new HashSet<>();
+    }
+
+    public Set<StubInfo> getStubs() {
+        return stubs;
+    }
+
+    public void addStub(DotName stubClass, StubType type) {
+        stubs.add(new StubInfo(stubClass, type));
+    }
+
+    public String getServiceName() {
+        return name;
+    }
+
+    public static final class StubInfo {
+
+        public final DotName className;
+        public final StubType type;
+
+        public StubInfo(DotName className, StubType type) {
+            this.className = className;
+            this.type = type;
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(className);
+        }
+
+        @Override
+        public boolean equals(Object obj) {
+            if (this == obj) {
+                return true;
+            }
+            if (obj == null) {
+                return false;
+            }
+            if (getClass() != obj.getClass()) {
+                return false;
+            }
+            StubInfo other = (StubInfo) obj;
+            return Objects.equals(className, other.className);
+        }
+
+    }
+
+    public enum StubType {
+
```

Comment:
maybe this would be a tiny bit better
```java
        BLOCKING("newBlockingStub"),
        MUTINY("newMutinyStub");

        private final factoryMethodName;
        StubType(String factoryMethod) {
                factoryMethodName = factoryMethod;
        }

        public String getFactoryMethodName() {
                return factoryMethodName;
        }
    }
```